includes/ProductSets/Sync.php

```diff
@@ -117,8 +117,12 @@ public function __construct() {
 	public function add_hooks() {
 
 		// product hooks, compare taxonomies the lists before and after saving product to see if must sync
-		add_filter( 'wp_insert_post_data', array( $this, 'filter_product_data' ), 99, 2 );
-		add_action( 'save_post', array( $this, 'maybe_sync_product_sets' ), 99, 2 );
+		add_filter( 'wp_insert_post_data', array( $this, 'check_product_data_before_save' ), 99, 2 );
+		add_action( 'save_post', array( $this, 'check_product_data_after_save' ), 99, 2 );//maybe_sync_product_sets
+
+		// product hooks, sync product's product set when deleting or restoring product
+		add_action( 'trashed_post', array( $this, 'sync_product_product_sets' ), 99 );
```

Comment:
Maybe you should also add a hook to sync on `before_delete_post` ?

Reply:
Added. Thanks.